boards/DIM/Src/App/states/App_DriveState.c

```diff
@@ -3,6 +3,95 @@
+}
+
+static bool App_DriveState_HasBMSWarning()
+{
+    return App_CanRx_BMS_Warnings_StackWatermarkAboveThresholdTask1Hz_Get() ||
+           App_CanRx_BMS_Warnings_StackWatermarkAboveThresholdTask100Hz_Get() ||
+           App_CanRx_BMS_Warnings_StackWatermarkAboveThresholdTask1kHz_Get() ||
+           App_CanRx_BMS_Warnings_StackWatermarkAboveThresholdTaskCanRx_Get() ||
+           App_CanRx_BMS_Warnings_StackWatermarkAboveThresholdTaskCanTx_Get() ||
+           App_CanRx_BMS_Warnings_WatchdogTimeout_Get();
+}
+
+static bool App_DriveState_HasDCMWarning()
+{
+    return App_CanRx_DCM_Warnings_StackWatermarkAboveThresholdTask1Hz_Get() ||
+           App_CanRx_DCM_Warnings_StackWatermarkAboveThresholdTask100Hz_Get() ||
+           App_CanRx_DCM_Warnings_StackWatermarkAboveThresholdTask1kHz_Get() ||
+           App_CanRx_DCM_Warnings_StackWatermarkAboveThresholdTaskCanRx_Get() ||
+           App_CanRx_DCM_Warnings_StackWatermarkAboveThresholdTaskCanTx_Get() ||
+           App_CanRx_DCM_Warnings_WatchdogTimeout_Get();
+}
+
+static bool App_DriveState_HasDIMWarning()
+{
+    return App_CanTx_DIM_Warnings_StackWatermarkAboveThresholdTask1Hz_Get() ||
+           App_CanTx_DIM_Warnings_StackWatermarkAboveThresholdTask100Hz_Get() ||
+           App_CanTx_DIM_Warnings_StackWatermarkAboveThresholdTask1kHz_Get() ||
+           App_CanTx_DIM_Warnings_StackWatermarkAboveThresholdTaskCanRx_Get() ||
+           App_CanTx_DIM_Warnings_StackWatermarkAboveThresholdTaskCanTx_Get() ||
+           App_CanTx_DIM_Warnings_WatchdogTimeout_Get();
+}
+
+static bool App_DriveState_HasFSMWarning()
+{
+    return App_CanRx_FSM_Warnings_PappsOutOfRange_Get() || App_CanRx_FSM_Warnings_SappsOutOfRange_Get() ||
+           App_CanRx_FSM_Warnings_StackWatermarkAboveThresholdTask1Hz_Get() ||
+           App_CanRx_FSM_Warnings_StackWatermarkAboveThresholdTask100Hz_Get() ||
+           App_CanRx_FSM_Warnings_StackWatermarkAboveThresholdTask1kHz_Get() ||
+           App_CanRx_FSM_Warnings_StackWatermarkAboveThresholdTaskCanRx_Get() ||
+           App_CanRx_FSM_Warnings_StackWatermarkAboveThresholdTaskCanTx_Get() ||
+           App_CanRx_FSM_Warnings_WatchdogTimeout_Get() || App_CanRx_FSM_Warnings_BrakeAccDisagreement_Get() ||
+           App_CanRx_FSM_Warnings_SteeringAngleSensorOCSC_Get() ||
+           App_CanRx_FSM_Warnings_LeftWheelSpeedOutOfRange_Get() != RANGE_CHECK_OK ||
+           App_CanRx_FSM_Warnings_RightWheelSpeedOutOfRange_Get() != RANGE_CHECK_OK ||
+           App_CanRx_FSM_Warnings_SteeringAngleOutOfRange_Get() != RANGE_CHECK_OK ||
+           App_CanRx_FSM_Warnings_FrontBrakePressureOutOfRange_Get() != RANGE_CHECK_OK ||
+           App_CanRx_FSM_Warnings_RearBrakePressureOutOfRange_Get() != RANGE_CHECK_OK ||
+           App_CanRx_FSM_Warnings_FlowRateOutOfRange_Get() != RANGE_CHECK_OK ||
+           App_CanRx_FSM_Warnings_PappsOCSCIsActive_Get() || App_CanRx_FSM_Warnings_SappsOCSCIsActive_Get() ||
+           App_CanRx_FSM_Warnings_FlowMeterHasUnderflow_Get();
+}
```

Comment:
Wow you're right this does suck

Reply:
We should really rethink this

Reply:
@Lucien950 Are these all of the FSM warning conditions? This function is reading all of the warning messages from CAN and returns true if any warning condition is true. 

Most of the other boards just have flags, I assumed that for the out-of-range signals that they would trigger a warning if the range check was not okay, is this correct?